### PR TITLE
Fix `-spec(_).`s

### LINKED ISSUE #8: Pursue `erl_opt` `warn_missing_spec` later

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
# Description

Closes #8.

This depends on https://github.com/arizona-framework/arizona/pull/104, so we'll have to re-run CI later.

But the changes are already Ok to get a sense of shape and what we want with Arizona consumption.

- [x] I have performed a self-review of my changes
- [x] I have read and understood the [contributing guidelines](/arizona-framework/arizona_example/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: src/arizona_example_live_counter.erl
Comment (paulo-ferraz-oliveira) [inline src/arizona_example_live_counter.erl:1]: We use a few newer getters/setters, types and the updated macro (which takes Macros as input).

**Review by williamthome — APPROVED** [review]

**Comment by williamthome** [comment]
Hmm, https://github.com/arizona-framework/arizona/pull/104 has been merged, but CI is still failing. Is it a cache bug? Or am I missing something?

**Comment by paulo-ferraz-oliveira** [comment]
It's not a cache bug. I either forget to push something or changed something last minute. I'll look at it 😄

**Review by williamthome — APPROVED** [review]
:monocle_face:

**Pull request merged**